subsys/dfu/dfu_target/src/dfu_target_stream.c

```diff
@@ -54,12 +54,27 @@ static int settings_set(const char *key, size_t len_rd,
 			settings_read_cb read_cb, void *cb_arg)
 {
 	if (!strcmp(key, current_id)) {
+		int err;
+		struct flash_pages_info page;
 		ssize_t len = read_cb(cb_arg, &stream.bytes_written,
 				      sizeof(stream.bytes_written));
+
 		if (len != sizeof(stream.bytes_written)) {
 			LOG_ERR("Can't read stream.bytes_written from storage");
 			return len;
 		}
+
+		err = flash_get_page_info_by_offs(stream.fdev,
+						  stream.bytes_written, &page);
+		if (err != 0) {
+			LOG_ERR("Error %d while getting page info", err);
+			return err;
+		}
+
+		/* Update the last erased page to avoid deleting already
+		 * written data.
+		 */
+		stream.last_erased_page_start_offset = page.start_offset;
```

Comment:
I guess this needs to be ifdefed since the last_erased_page_start_offset element is ifdefed.
```suggestion
#ifdef CONFIG_STREAM_FLASH_ERASE
		err = flash_get_page_info_by_offs(stream.fdev,
						  stream.bytes_written, &page);
		if (err != 0) {
			LOG_ERR("Error %d while getting page info", err);
			return err;
		}

		/* Update the last erased page to avoid deleting already
		 * written data.
		 */
		stream.last_erased_page_start_offset = page.start_offset;
#endif
```

Reply:
Good spot! Thanks, will try using IS_ENABLED instead of ifdef directly.

Reply:
Actually, `dfu_target_stream` has a dependency to 

```
config DFU_TARGET_STREAM
	bool "Generic DFU stream target"
	depends on STREAM_FLASH_ERASE
	depends on STREAM_FLASH
```

So this would not be necessary